content/sensu-core/2.0/reference/agent.md

```diff
@@ -0,0 +1,803 @@
+At the core of this model are Sensu agent subscriptions.
+
+Each Sensu agent has a defined set of [`subscriptions`][28], a list of roles and responsibilities assigned to the system (for example: for a webserver or database).
+These subscriptions determine which [monitoring checks][14] are executed by the agent.
+Agent subscriptions allow Sensu to request check executions on a group of systems at a time, instead of a traditional 1:1 mapping of configured hosts to monitoring checks.
+Sensu checks target Sensu agent subscriptions, using the [check definition attribute `subscriptions`][14].
+
+After receiving the check request from the Sensu backend, the agent:
+
+1. Applies any [tokens][27] matching attribute values in the check definition.
+2. Fetches [assets][29] and stores them in its local cache. By default, agents cache asset data at `/var/cache/sensu/sensu-agent` (`C:\\ProgramData\sensu\cache\sensu-agent` on Windows systems) or as specified by the the [`--cache-dir` flag][30].
+3. Executes the [check `command`][14].
+4. Executes any [hooks][31] specified by the check based on the exit status.
+5. Creates an [event][7] containing information about the applicable entity, check, and metric.
+
+### Subscription configuration
+
+In order for an agent to execute a service check, you must specify the same subscription in the [agent configuration][24] and the [check configuration][32].
+To configure subscriptions for an agent, see [the `subscriptions` flag reference][28].
+
+In addition to the subscriptions defined in the agent configuration, Sensu agents also subscribe automatically to a subscription matching their agent `id`.
+For example, an agent with the ID `i-424242` will subscribe to check requests with the subscription `agent:i-424242`.
+This makes it possible to generate ad-hoc check requests targeting specific clients via the API.
+
+### Proxy entities
+
+Sensu proxy entities allow Sensu to monitor external resources on systems or devices where a Sensu agent cannot be installed (such a network switch) using the defined [check `proxy_entity_id`][14] to create a proxy entity for the external resource.
+See the [entity reference documentation][3] and the [guide to monitoring external resources][33] for more information about proxy entities.
+
+## Creating events using the agent socket
+
+Every Sensu agent has a TCP, UDP, and HTTP socket listening for external check result input.
+The TCP and UDP sockets listen on the address and port specified by the [socket configuration flags][17]; the HTTP socket listens on the address and port specified by the [API configuration flags][18].
+
+These sockets expect JSON formatted check results, allowing external sources (such as your web application or backup scripts) to push check results without needing to know anything about Sensu's internal implementation.
+An excellent agent socket use case example is a web application pushing check results to indicate database connectivity issues.
```

Comment:
The line above this one provides the start of two examples in `()`. Maybe chose one spot to cut down on repetition. I like this line better.